### ISSUE TITLE
Adding lambda layers to functions broke my amplify stack

### ISSUE THREAD

**Issue by kylekirkby** [aws-amplify/amplify-cli#4932]
**Describe the bug**
I thought I'd try and share the helper JS files across my lambda functions but am now faced with a long list of errors:
```
Embedded stack arn:aws:cloudformation:us-east-1:xxxxxxx:stack/amplify-xxxxxxxx-develop-200029-functiongetScreenshot-xxxxxx/aaadfdd0-83ff-11ea-a49e-0ad4194c8c9d was not successfully updated. Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: No export named apixxxxxxGraphQLAPIIdOutput:GetAtt:ImageTable:Nam
```

For reference, I've removed the changes that I made that caused this issue and am now stuck in this failed stack state...

```
UPDATE_FAILED      AmplifyResourcesPolicy AWS::IAM::Policy      Fri Jul 24 2020 17:39:18 GMT+0100 (British Summer Time) The policy failed legacy parsing (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: xxxxxx-430f-a58c-e66e565ee4c1)
```

I get around 10 of these..

**Amplify CLI Version**
4.25.0

**To Reproduce**
Add a lambda layer to your functions. Add some js files to your opt/ folder.

**Expected behavior**
Adding a lambda layer shouldn't break the stack.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
- Node: 12

**Additional context**
This may be something to do with enabling analytics permissions in one of my functions ... (I've seen others in the issues having similar problems). Not sure where to go now...updating the stack takes a while and without removing the lamber layers from each function one by one, I don't think I'll get to the bottom of the issue.

**Comment by kylekirkby** [comment]
Looks like it could be something to do with the modifications that were made to the backend config...

```
-            "dependsOn": [
-                {
-                    "category": "auth",
-                    "resourceName": "xxxxxappa2ad5d26",
-                    "attributes": [
-                        "UserPoolId"
-                    ]
-                },
-                {
-                    "category": "api",
-                    "resourceName": "xxxxxxx",
-                    "attributes": [
-                        "GraphQLAPIIdOutput",
-                        "GraphQLAPIEndpointOutput"
-                    ]
-                },
-                {
-                    "category": "storage",
-                    "resourceName": "userAssets",
-                    "attributes": [
-                        "BucketName"
-                    ]
-                }
-            ]
+            "dependsOn": []

```

When doing a git diff all of my dependencies for functions have been removed.

**Comment by jhockett** [comment]
Related: https://github.com/aws-amplify/amplify-cli/issues/4852

**Comment by jhockett** [comment]
A fix was merged and deployed in Amplify CLI version 4.26.0.
